src/providers/auth.ts

```diff
@@ -74,33 +75,40 @@ export class FirebaseAuth extends ReplaySubject<FirebaseAuthState> {
 
     switch (config.method) {
       case AuthMethods.Popup:
-        return this._authBackend.authWithOAuthPopup(config.provider, this._scrubConfig(config));
+        return this._authBackend.authWithOAuthPopup(config.provider, this._scrubConfig(config))
+          .then((userCredential: firebase.auth.UserCredential) => {
+            return authDataToAuthState(userCredential.user, <OAuthCredential>(<any>userCredential).credential)
+          });
       case AuthMethods.Redirect:
-        return this._authBackend.authWithOAuthRedirect(config.provider, this._scrubConfig(config));
+        // Gets around typings issue since this method doesn't resolve with a user.
+        // The method really only does anything with an error, since it redirects.
+        return <Promise<FirebaseAuthState>>(<any>this._authBackend).authWithOAuthRedirect(config.provider, this._scrubConfig(config));
       case AuthMethods.Anonymous:
         return this._authBackend.authAnonymously(this._scrubConfig(config));
       case AuthMethods.Password:
-        return this._authBackend.authWithPassword(<FirebaseCredentials>credentials, this._scrubConfig(config, false));
+        return this._authBackend.authWithPassword(<EmailPasswordCredentials>credentials);
       case AuthMethods.OAuthToken:
-        return this._authBackend.authWithOAuthToken(config.provider, <OAuthCredentials>credentials,
+        return this._authBackend.authWithOAuthToken(<firebase.auth.AuthCredential>credentials,
           this._scrubConfig(config));
       case AuthMethods.CustomToken:
-        return this._authBackend.authWithCustomToken((<OAuth2Credentials>credentials).token,
-          this._scrubConfig(config, false));
+        return this._authBackend.authWithCustomToken(<AuthToken>credentials);
     }
   }
 
   public logout(): void {
-    if (this._authBackend.getAuth() !== null) {
-      this._authBackend.unauth();
-    }
+    this._authBackend.unauth();
   }
 
-  public getAuth(): FirebaseAuthData {
-    return this._authBackend.getAuth();
+  public getAuth(): FirebaseAuthState {
+    return this._authBackend.getAuth()
   }
 
-  public createUser(credentials: FirebaseCredentials): Promise<FirebaseAuthData> {
+  //TODO: Make breaking change note
```

Comment:
This method is not tested and not necessary since the service is itself observable.